src/frontend.rs

```diff
@@ -72,6 +82,27 @@ pub trait SearchWithLocate<'a, T>: Search<'a, T> {
     fn locate(&self) -> Vec<u64>;
 }
 
+/// A match in the text.
+pub trait Match<'a, T> {
+    /// Iterate over the characters of the match.
+    fn iter_chars_forward(&self) -> impl Iterator<Item = T> + 'a;
+
+    /// Iterate over the characters of the match in reverse.
+    fn iter_chars_backward(&self) -> impl Iterator<Item = T> + 'a;
```

Comment:
We may consider deleting `Search::iter_forward(&'a self, i: u64)` and `Search::iter_backward(&'a self, i: u64)` as these new methods offer the same feature.